app/models/container_service.rb

```diff
@@ -27,10 +27,23 @@ def container_groups_count
     number_of(:container_groups)
   end
 
+  include EventMixin
   include Metric::CiMixin
 
   PERF_ROLLUP_CHILDREN = :container_groups
 
+  def event_where_clause(assoc = :ems_events)
+    case assoc.to_sym
+    when :ems_events
+      # TODO: improve relationship using the id
+      ["container_namespace = ? AND container_group_name IN (?) AND #{events_table_name(assoc)}.ems_id = ?",
```

Comment:
@dkorn this is not the way to go. We have to do the service association when we receive the event, not when we display them. Which means that you have to add at least a `container_service_id` column to `events_stream` and have the Catcher/Parser do the association. Here we just display it.